### ISSUE TITLE
Installing dede on Jetson TX2

### ISSUE THREAD

**Issue by zakhariarestu** [jolibrain/deepdetect#516]
#### Configuration
- Version of DeepDetect:
    - Locally compiled on:
        - Ubuntu 16.04 LTS
        - NVIDIA Jetson TX2
- Commit (shown by the server when starting):

#### Your question / the problem you're facing:
How to fix this problem? I get some error when I execute "make"  on my jetson TX2

I've  done --> cmake .. -DCUDA_ARCH="-gencode arch=compute_62,code=sm_62" -DUSE_CUDNN=ON -DJETSON=ON -DCUDA_USE_STATIC_CUDA_RUNTIME=OFF

#### Error message (if any) / steps to reproduce the problem:
- Server log output:
nvidia@tegra-ubuntu:~/deepdetect/build$ sudo make
[  3%] Creating directories for 'caffe_dd'
[  7%] Performing download step (download, verify and extract) for 'caffe_dd'
-- downloading...
     src='https://github.com/jolibrain/caffe/archive/master.tar.gz'
     dst='/home/nvidia/deepdetect/build/caffe_dd/src/master.tar.gz'
     timeout='none'
-- downloading... done
-- verifying file...
     file='/home/nvidia/deepdetect/build/caffe_dd/src/master.tar.gz'
-- verifying file... warning: did not verify file - no URL_HASH specified?
-- extracting...
     src='/home/nvidia/deepdetect/build/caffe_dd/src/master.tar.gz'
     dst='/home/nvidia/deepdetect/build/caffe_dd/src/caffe_dd'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 10%] No patch step for 'caffe_dd'
[ 14%] No update step for 'caffe_dd'
[ 17%] Performing configure step for 'caffe_dd'
PROTOC src/caffe/proto/caffe.proto
CXX .build_release/src/caffe/proto/caffe.pb.cc
CXX src/caffe/util/sampler.cpp
CXX src/caffe/util/io.cpp
CXX src/caffe/util/blocking_queue.cpp
CXX src/caffe/util/hdf5.cpp
CXX src/caffe/util/db_leveldb.cpp
CXX src/caffe/util/db_lmdb.cpp
CXX src/caffe/util/bbox_util.cpp
src/caffe/util/io.cpp: In function ‘void caffe::CVMatToDatum(const cv::Mat&, caffe::Datum*)’:
src/caffe/util/io.cpp:796:9: warning: unused variable ‘datum_size’ [-Wunused-variable]
     int datum_size = datum_channels * datum_height * datum_width;
         ^
CXX src/caffe/util/benchmark.cpp
CXX src/caffe/util/cudnn.cpp
CXX src/caffe/util/im_transforms.cpp
CXX src/caffe/util/signal_handler.cpp
CXX src/caffe/util/db.cpp
CXX src/caffe/util/upgrade_proto.cpp
CXX src/caffe/util/interp.cpp
CXX src/caffe/util/im2col.cpp
CXX src/caffe/util/insert_splits.cpp
src/caffe/util/upgrade_proto.cpp: In function ‘bool caffe::UpgradeV1LayerParameter(const caffe::V1LayerParameter&, caffe::LayerParameter*)’:
src/caffe/util/upgrade_proto.cpp:721:26: warning: ‘mode’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   ParamSpec_DimCheckMode mode;
                          ^
CXX src/caffe/util/math_functions.cpp
CXX src/caffe/layers/memory_data_layer.cpp
CXX src/caffe/layers/detection_output_layer.cpp
CXX src/caffe/layers/sigmoid_cross_entropy_loss_layer.cpp
CXX src/caffe/layers/neuron_layer.cpp
CXX src/caffe/layers/power_layer.cpp
src/caffe/layers/sigmoid_cross_entropy_loss_layer.cpp: In member function ‘Dtype caffe::SigmoidCrossEntropyLossLayer<Dtype>::get_normalizer(caffe::LossParameter_NormalizationMode, int) [with Dtype = float]’:
src/caffe/layers/sigmoid_cross_entropy_loss_layer.cpp:75:41: warning: ‘normalizer’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   return std::max(Dtype(1.0), normalizer);
                                         ^
src/caffe/layers/sigmoid_cross_entropy_loss_layer.cpp: In member function ‘Dtype caffe::SigmoidCrossEntropyLossLayer<Dtype>::get_normalizer(caffe::LossParameter_NormalizationMode, int) [with Dtype = double]’:
src/caffe/layers/sigmoid_cross_entropy_loss_layer.cpp:75:41: warning: ‘normalizer’ may be used uninitialized in this function [-Wmaybe-uninitialized]
CXX src/caffe/layers/lstm_unit_layer.cpp
CXX src/caffe/layers/swish_layer.cpp
CXX src/caffe/layers/selu_layer.cpp
CXX src/caffe/layers/prior_box_layer.cpp
CXX src/caffe/layers/split_layer.cpp
CXX src/caffe/layers/im2col_layer.cpp
CXX src/caffe/layers/smooth_l1_loss_layer.cpp
CXX src/caffe/layers/euclidean_loss_layer.cpp
CXX src/caffe/layers/dense_image_data_layer.cpp
CXX src/caffe/layers/parameter_layer.cpp
CXX src/caffe/layers/cudnn_pooling_layer.cpp
CXX src/caffe/layers/continuation_indicator_layer.cpp
CXX src/caffe/layers/bnll_layer.cpp
CXX src/caffe/layers/batch_norm_layer.cpp
CXX src/caffe/layers/relu_layer.cpp
CXX src/caffe/layers/exp_layer.cpp
CXX src/caffe/layers/filter_layer.cpp
CXX src/caffe/layers/bias_layer.cpp
CXX src/caffe/layers/scale_layer.cpp
CXX src/caffe/layers/normalize_layer.cpp
CXX src/caffe/layers/base_data_layer.cpp
CXX src/caffe/layers/deconv_layer.cpp
CXX src/caffe/layers/slice_layer.cpp
CXX src/caffe/layers/elu_layer.cpp
CXX src/caffe/layers/softmax_loss_layer.cpp
CXX src/caffe/layers/roi_pooling_layer.cpp
CXX src/caffe/layers/base_conv_layer.cpp
CXX src/caffe/layers/focal_loss_layer.cpp
CXX src/caffe/layers/permute_layer.cpp
CXX src/caffe/layers/sigmoid_layer.cpp
CXX src/caffe/layers/cudnn_sigmoid_layer.cpp
CXX src/caffe/layers/batch_reindex_layer.cpp
CXX src/caffe/layers/cudnn_relu_layer.cpp
CXX src/caffe/layers/hinge_loss_layer.cpp
CXX src/caffe/layers/infogain_loss_layer.cpp
CXX src/caffe/layers/sparse_data_layer.cpp
CXX src/caffe/layers/clip_layer.cpp
CXX src/caffe/layers/ctc_loss_layer.cpp
CXX src/caffe/layers/dice_coef_loss_layer.cpp
CXX src/caffe/layers/concat_layer.cpp
CXX src/caffe/layers/multibox_loss_layer.cpp
CXX src/caffe/layers/image_data_layer.cpp
CXX src/caffe/layers/embed_layer.cpp
CXX src/caffe/layers/tile_layer.cpp
CXX src/caffe/layers/sparse_inner_product_layer.cpp
CXX src/caffe/layers/softmax_layer.cpp
CXX src/caffe/layers/conv_layer.cpp
CXX src/caffe/layers/selu_dropout_layer.cpp
CXX src/caffe/layers/argmax_layer.cpp
CXX src/caffe/layers/threshold_layer.cpp
CXX src/caffe/layers/reshape_layer.cpp
CXX src/caffe/layers/cross_entropy_loss.cpp
CXX src/caffe/layers/lrn_layer.cpp
CXX src/caffe/layers/dummy_data_layer.cpp
CXX src/caffe/layers/loss_layer.cpp
CXX src/caffe/layers/cudnn_deconv_layer.cpp
CXX src/caffe/layers/contrastive_loss_layer.cpp
CXX src/caffe/layers/recurrent_layer.cpp
src/caffe/layers/loss_layer.cpp: In member function ‘Dtype caffe::LossLayer<Dtype>::GetNormalizer(caffe::LossParameter_NormalizationMode, int, int, int) [with Dtype = float]’:
src/caffe/layers/loss_layer.cpp:54:41: warning: ‘normalizer’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   return std::max(Dtype(1.0), normalizer);
                                         ^
src/caffe/layers/loss_layer.cpp: In member function ‘Dtype caffe::LossLayer<Dtype>::GetNormalizer(caffe::LossParameter_NormalizationMode, int, int, int) [with Dtype = double]’:
src/caffe/layers/loss_layer.cpp:54:41: warning: ‘normalizer’ may be used uninitialized in this function [-Wmaybe-uninitialized]
CXX src/caffe/layers/shuffle_channel_layer.cpp
CXX src/caffe/layers/reduction_layer.cpp
CXX src/caffe/layers/interp_layer.cpp
CXX src/caffe/layers/window_data_layer.cpp
CXX src/caffe/layers/tanh_layer.cpp
CXX src/caffe/layers/softmax_infogain_loss_layer.cpp
CXX src/caffe/layers/data_layer.cpp
CXX src/caffe/layers/multi_label_accuracy_layer.cpp
src/caffe/layers/softmax_infogain_loss_layer.cpp: In member function ‘Dtype caffe::SoftmaxWithInfogainLossLayer<Dtype>::get_normalizer(caffe::LossParameter_NormalizationMode, int) [with Dtype = float]’:
src/caffe/layers/softmax_infogain_loss_layer.cpp:111:43: warning: ‘normalizer’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return std::max(Dtype(1.0), normalizer);
                                           ^
src/caffe/layers/softmax_infogain_loss_layer.cpp: In member function ‘Dtype caffe::SoftmaxWithInfogainLossLayer<Dtype>::get_normalizer(caffe::LossParameter_NormalizationMode, int) [with Dtype = double]’:
src/caffe/layers/softmax_infogain_loss_layer.cpp:111:43: warning: ‘normalizer’ may be used uninitialized in this function [-Wmaybe-uninitialized]
CXX src/caffe/layers/inner_product_layer.cpp
CXX src/caffe/layers/dropout_layer.cpp
CXX src/caffe/layers/cudnn_lcn_layer.cpp
CXX src/caffe/layers/detection_evaluate_layer.cpp
CXX src/caffe/layers/multi_label_sigmoid_loss_layer.cpp
CXX src/caffe/layers/silence_layer.cpp
CXX src/caffe/layers/spp_layer.cpp
CXX src/caffe/layers/rnn_layer.cpp
CXX src/caffe/layers/multibox_focal_loss_layer.cpp
CXX src/caffe/layers/accuracy_layer.cpp
CXX src/caffe/layers/video_data_layer.cpp
CXX src/caffe/layers/upsample_layer.cpp
CXX src/caffe/layers/lstm_layer.cpp
CXX src/caffe/layers/input_layer.cpp
CXX src/caffe/layers/labelsequence_accuracy_layer.cpp
CXX src/caffe/layers/hdf5_data_layer.cpp
CXX src/caffe/layers/absval_layer.cpp
CXX src/caffe/layers/memory_sparse_data_layer.cpp
CXX src/caffe/layers/cudnn_softmax_layer.cpp
CXX src/caffe/layers/annotated_data_layer.cpp
CXX src/caffe/layers/prelu_layer.cpp
CXX src/caffe/layers/cudnn_conv_layer.cpp
CXX src/caffe/layers/multinomial_logistic_loss_layer.cpp
CXX src/caffe/layers/axpy_layer.cpp
CXX src/caffe/layers/eltwise_layer.cpp
CXX src/caffe/layers/conv_dw_layer.cpp
CXX src/caffe/layers/hdf5_output_layer.cpp
CXX src/caffe/layers/flatten_layer.cpp
CXX src/caffe/layers/cudnn_lrn_layer.cpp
CXX src/caffe/layers/mvn_layer.cpp
CXX src/caffe/layers/log_layer.cpp
CXX src/caffe/layers/crop_layer.cpp
CXX src/caffe/layers/cudnn_tanh_layer.cpp
CXX src/caffe/layers/pooling_layer.cpp
CXX src/caffe/syncedmem.cpp
CXX src/caffe/net.cpp
CXX src/caffe/sparse_blob.cpp
CXX src/caffe/internal_thread.cpp
CXX src/caffe/layer_factory.cpp
CXX src/caffe/data_transformer.cpp
CXX src/caffe/common.cpp
CXX src/caffe/data_reader.cpp
CXX src/caffe/parallel.cpp
CXX src/caffe/3rdparty/ctc/ctc_entrypoint.cpp
In file included from src/caffe/3rdparty/ctc/ctc_entrypoint.cpp:7:0:
./include/caffe/3rdparty/ctc/detail/cpu_ctc.cuh: In instantiation of ‘CpuCTC<ProbT>::CpuCTC(int, int, void*, int, int) [with ProbT = float]’:
src/caffe/3rdparty/ctc/ctc_entrypoint.cpp:46:63:   required from here
./include/caffe/3rdparty/ctc/detail/cpu_ctc.cuh:77:11: warning: ‘CpuCTC<float>::workspace_’ will be initialized after [-Wreorder]
     void* workspace_;
           ^
./include/caffe/3rdparty/ctc/detail/cpu_ctc.cuh:76:9: warning:   ‘int CpuCTC<float>::blank_label_’ [-Wreorder]
     int blank_label_;
         ^
./include/caffe/3rdparty/ctc/detail/cpu_ctc.cuh:20:5: warning:   when initialized here [-Wreorder]
     CpuCTC(int alphabet_size, int minibatch, void* workspace, int num_threads,
     ^
src/caffe/3rdparty/ctc/ctc_entrypoint.cpp: In function ‘ctcStatus_t compute_ctc_loss(const float*, float*, const int*, const int*, const int*, int, int, float*, void*, ctcOptions)’:
src/caffe/3rdparty/ctc/ctc_entrypoint.cpp:90:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
CXX src/caffe/layer.cpp
CXX src/caffe/blob.cpp
CXX src/caffe/solvers/sgd_solver.cpp
CXX src/caffe/solvers/adagrad_solver.cpp
CXX src/caffe/solvers/nesterov_solver.cpp
CXX src/caffe/solvers/adadelta_solver.cpp
src/caffe/solvers/sgd_solver.cpp: In member function ‘Dtype caffe::SGDSolver<Dtype>::GetLearningRate() [with Dtype = float]’:
src/caffe/solvers/sgd_solver.cpp:112:10: warning: ‘rate’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   return rate;
          ^
src/caffe/solvers/sgd_solver.cpp: In member function ‘Dtype caffe::SGDSolver<Dtype>::GetLearningRate() [with Dtype = double]’:
src/caffe/solvers/sgd_solver.cpp:112:10: warning: ‘rate’ may be used uninitialized in this function [-Wmaybe-uninitialized]
CXX src/caffe/solvers/adam_solver.cpp
CXX src/caffe/solvers/rmsprop_solver.cpp
src/caffe/blob.cpp: In member function ‘Dtype caffe::Blob<Dtype>::sumsq_data() const [with Dtype = float]’:
src/caffe/blob.cpp:290:9: warning: ‘sumsq’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   Dtype sumsq;
         ^
src/caffe/blob.cpp: In member function ‘Dtype caffe::Blob<Dtype>::sumsq_diff() const [with Dtype = float]’:
src/caffe/blob.cpp:333:9: warning: ‘sumsq’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   Dtype sumsq;
         ^
src/caffe/blob.cpp: In member function ‘Dtype caffe::Blob<Dtype>::sumsq_data() const [with Dtype = double]’:
src/caffe/blob.cpp:290:9: warning: ‘sumsq’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   Dtype sumsq;
         ^
src/caffe/blob.cpp: In member function ‘Dtype caffe::Blob<Dtype>::sumsq_diff() const [with Dtype = double]’:
src/caffe/blob.cpp:333:9: warning: ‘sumsq’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   Dtype sumsq;
         ^
CXX src/caffe/solver.cpp
CXX tools/upgrade_net_proto_text.cpp
CXX tools/finetune_net.cpp
CXX tools/upgrade_solver_proto_text.cpp
CXX tools/get_image_size.cpp
CXX tools/device_query.cpp
CXX tools/create_label_map.cpp
CXX tools/convert_annoset.cpp
CXX tools/extract_features.cpp
CXX tools/caffe.cpp
CXX tools/train_net.cpp
CXX tools/convert_imageset.cpp
CXX tools/net_speed_benchmark.cpp
CXX tools/upgrade_net_proto_binary.cpp
tools/caffe.cpp: In function ‘caffe::SolverAction::Enum GetRequestedAction(const string&)’:
tools/caffe.cpp:177:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
CXX tools/test_net.cpp
CXX tools/compute_image_mean.cpp
CXX examples/cifar10/convert_cifar_data.cpp
CXX examples/ssd/ssd_detect.cpp
CXX examples/mnist/convert_mnist_data.cpp
CXX examples/siamese/convert_mnist_siamese_data.cpp
CXX examples/warpctc_captcha/captcha_prediction.cpp
CXX examples/cpp_classification/classification.cpp
AR -o .build_release/lib/libcaffe.a
LD -o .build_release/lib/libcaffe.so.1.0.0-rc3
/usr/bin/ld: cannot find -lhdf5_hl
/usr/bin/ld: cannot find -lhdf5
collect2: error: ld returned 1 exit status
Makefile:597: recipe for target '.build_release/lib/libcaffe.so.1.0.0-rc3' failed
make[3]: *** [.build_release/lib/libcaffe.so.1.0.0-rc3] Error 1
make[3]: *** Waiting for unfinished jobs....
CMakeFiles/caffe_dd.dir/build.make:108: recipe for target 'caffe_dd/src/caffe_dd-stamp/caffe_dd-configure' failed
make[2]: *** [caffe_dd/src/caffe_dd-stamp/caffe_dd-configure] Error 2
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/caffe_dd.dir/all' failed
make[1]: *** [CMakeFiles/caffe_dd.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

Thanks

**Comment by zakhariarestu** [comment]
thanks beniz, i got it, work perfectly

**Comment by beniz** [comment]
Good to hear, have fun! Let us know of embedded use cases we could help with.
